client/blocks/reader-related-card-v2/docs/example.jsx

```diff
@@ -56,24 +56,76 @@ const smallItems = [
+	{
+		post: {
+			ID: 3,
+			global_ID: 3,
+			site_ID: 3,
+			author: {
+				name: "ConservativeJoe",
+			},
+			date: "2016-09-27T08:20:33+00:00",
+			modified: "2016-09-27T08:20:37+00:00",
+			title: "Why Trump should Win the American Election.",
+			reading_time: 152,
+			better_excerpt: "This presidential race has, to be blunt, a cluster bomb of attacks, lies and outright nonsense rhetoric that has astounded many and made people curse at their screens, newspapers and media agencies they follow.",
+			short_excerpt: "This presidential race has, to be blunt, a cluster bomb of attacks, lies and outright nonsense rhetoric that has astounded many and made people curse at their…",
+			canonical_image: {
+					uri: "https://freedomsandtruth.files.wordpress.com/2016/09/trump-rally-in-vegas-getty-640x480.jpg?w=720&quality=80&strip=info",
+			}
+		},
+		site: {
+			title: 'All the catsss'
+		}
+	},
+	{
+		post: {
+			ID: 4,
+			global_ID: 4,
+			site_ID: 4,
+			author: {
+				name: "LiberalJoe",
+			},
+			date: "2016-09-27T08:20:33+00:00",
+			modified: "2016-09-27T08:20:37+00:00",
+			title: "Why Clinton should Win the American Election.",
+			reading_time: 152,
+			better_excerpt: "This presidential race, has, to be blunt, a lorem ipsum of devdocs demos.  HRC deserves the win because shes been playing the game longer than anybody, especially longer than trump. And she's tough.",
+			short_excerpt: "This presidential race has, to be blunt, a cluster bomb of attacks, lies and outright nonsense rhetoric that has astounded many and made people curse at their…",
+			canonical_image: {
+					uri: "https://lh4.googleusercontent.com/-eXKU4UhFusI/AAAAAAAAAAI/AAAAAAAAATA/1QahWqsqd-I/s0-c-k-no-ns/photo.jpg",
+			}
+		},
+		site: {
+			title: '99 Problems'
+		}
+	},
+];
+
 const RelatedPostCardv2Example = () => (
 	<div className="design-assets__group">
 		<Card>
-			<div className="reader-related-card-v2__container">
-				<h1 className="reader-related-card-v2__heading">More in <a href="#" className="reader-related-card-v2__link">Cats and Cats</a></h1>
-				<div className="reader-related-card-v2__posts is-site-post">
-					{ smallItems.map( item => <RelatedPostCard key={ item.post.global_ID } post={ item.post } site={ item.site } /> ) }
-				</div>
-			</div>
-		</Card>
-
-		<Card>
-			<div className="reader-related-card-v2__container">
-				<h1 className="reader-related-card-v2__heading">More in <a href="#" className="reader-related-card-v2__link">WordPress.com</a></h1>
-				<div className="reader-related-card-v2__posts is-global-post">
-					{ smallItems.map( item => <RelatedPostCard key={ item.post.global_ID } post={ item.post } site={ item.site } /> ) }
-				</div>
-			</div>
+		<div className="reader-related-card-v2__blocks is-same-site">
+			<h1 className="reader-related-card-v2__heading">More in <a className="reader-related-card-v2__link">Cats and Furballs</a></h1>
+				<ul className="reader-related-card-v2__list">
+					{ moreOnSameSite.map( item =>
```

Comment:
Either parens or braces work, just with their own caveats!
Using curlies would require you to explicitly return the component. 
However when using parens you don't since the code within them will just be evaluated and returned implicitly to the arrow function.  

So these will both essentially do the same thing:

```javascript
moreOnSameSite.map( item => {
    return <Component /> // explict return
} )
```

```javascript
moreOnSameSite.map( item => (
    <Component /> // implicit return 
) )
```

As a rule of thumb I typically use braces when some sort of extra logic or or variable declarations need to happen within the block before returning the component/value and use parens when the value can just be returned implicitly 

[Here's a good recent example](https://github.com/Automattic/wp-calypso/pull/8499/files#diff-da15028af77d9b929a0a8e262e985c15R128) of the paren usage

This has got me thinking though... the first example labelled as 'bad' uses parens for execution but claims that doing so is 'too magic'... Since it seems to be a fairly common pattern for returning components it might be worth revising or adding another example to the guidelines! I'll get on that and link the PR here :D